core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,388 @@
+    return {
+      data: [this.createTranslationReviewStat(stat)],
+      language: this.languageUtilService.getAudioLanguageDescription(
+        stat.language_code),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
+    };
+  }
+
+  createQuestionContributionStatsPage(
+      stats: QuestionContributionBackendDict[]): StatsPage {
+    return {
+      data: stats.map((stat) => {
+        return this.createQuestionContributionStat(stat);
+      }),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
+    };
+  }
+
+  createQuestionReviewStatsPage(
+      stat: QuestionReviewBackendDict[]): StatsPage {
+    return {
+      data: stat.map((stat) => {
+        return this.createQuestionReviewStat(stat);
+      }),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
+    };
+  }
+
+  createTranslationContributionStat(
+      stat: TranslationContributionBackendDict): TranslationContributionStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      acceptedCards: stat.accepted_translations_count,
+      acceptedWordCount: stat.accepted_translation_word_count
+    };
+  }
+
+  createTranslationReviewStat(
+      stat: TranslationReviewBackendDict): TranslationReviewStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      acceptedCards: stat.accepted_translations_count,
+      acceptedWordCount: stat.accepted_translation_word_count,
+      reviewedCards: stat.reviewed_translations_count,
+      reviewedWordCount: stat.reviewed_translation_word_count
+    };
+  }
+
+  createQuestionContributionStat(
+      stat: QuestionContributionBackendDict): QuestionContributionStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      acceptedQuestions: stat.accepted_questions_count,
+      acceptedQuestionsWithoutEdits: (
+        stat.accepted_questions_without_reviewer_edits_count)
+    };
+  }
+
+  createQuestionReviewStat(
+      stat: QuestionReviewBackendDict): QuestionReviewStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      reviewedQuestions: stat.reviewed_questions_count,
+      acceptedQuestions: stat.accepted_questions_count
+    };
+  }
+
+  goToNextPage(page: StatsPage): void {
+    page.currentPageStartIndex += this.ITEMS_PER_PAGE;
+    page.currentPageEndIndex += this.ITEMS_PER_PAGE;
+  }
+
+  goToPreviousPage(page: StatsPage): void {
+    page.currentPageStartIndex -= this.ITEMS_PER_PAGE;
+    page.currentPageEndIndex -= this.ITEMS_PER_PAGE;
+  }
+
+  getColumnSortDirection(): number {
+    return 0;
```

Comment:
It's not clear what 0 means, please explain. Perhaps you should use an enum? At any rate, this needs clear code comments.

Reply:
Done